### DISCUSSION TITLE
Migrations like Ruby on Rails

### DISCUSSION THREAD

**Discussion by TheBlckbird** [laravel/framework#48123] [Ideas]
Rails uses a system where they only have a `change` function in the migrations and the migrator can automatically roll back actions like creating a column or index. You can also optionally use an `up` and `down` method or a special function inside the `change` function for actions that the migrator can't roll back automatically.
[Look at the Rails docs for more information](https://guides.rubyonrails.org/active_record_migrations.html)

This should be fairly easy to implement, and I could maybe do it if desired.

**Comment by Sophist-UK** [comment]
Would it be backwards compatible with the existing implementation?

**Reply by TheBlckbird** [reply]
yes, as I said you could still use the old system
Have a look at the first section in the Rails docs, it explains a lot
